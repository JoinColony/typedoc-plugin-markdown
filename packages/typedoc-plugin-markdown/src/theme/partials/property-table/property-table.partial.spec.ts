import { ProjectReflection } from 'typedoc';
import { propertyTablePartial } from './property-table.partial';

describe(`Property Table Partial:`, () => {
  let project: ProjectReflection;

  beforeAll(async () => {
    project = global.getProject('declarations.ts');
  });

  test(`should compile property table for type literal`, () => {
    expect(
      propertyTablePartial(
        global.getMockContext(),
        (project.findReflectionByName('typeLiteralDeclaration') as any).type
          ?.declaration?.children,
      ),
    ).toMatchSnapshot();
  });
});
