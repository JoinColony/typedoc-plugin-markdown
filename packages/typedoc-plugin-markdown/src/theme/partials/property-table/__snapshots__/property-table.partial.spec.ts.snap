// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Table Partial: should compile property table for type literal 1`] = `
"
| Name | Type | Description |
| :------ | :------ | :------ |
| \`valueA?\` | \`number\` | Comments for valueA |
| \`valueB?\` | \`boolean\` | - |
| \`valueC\` | { \`[dataId: string]\`: \`\`\\"ok\\"\`\` \\\\| \`\`\\"ko\\"\`\`;  } | - |
| \`valueX\` | { \`valueA\`: \`number\`[] ; \`valueY\`: (\`z\`: \`string\`) => { \`a\`: \`string\` ; \`b\`: \`string\`  } ; \`valueZ\`: \`string\`  } | Comment for valueX |
| \`valueX.valueA\` | \`number\`[] | - |
| \`valueX.valueY\` | (\`z\`: \`string\`) => { \`a\`: \`string\` ; \`b\`: \`string\`  } | - |
| \`valueX.valueZ\` | \`string\` | Nested comment for valueZ |
| \`valueY\` | () => \`string\` | - |
| \`valueZ\` | \`string\` | Comment for valueZ |
"
`;
